packages/zoe/src/zoeService/zoe.js

```diff
@@ -44,37 +43,80 @@ import { ZoeServiceIKit } from '../typeGuards.js';
  * @param {Baggage} [zoeBaggage]
  */
 const makeZoeKit = (
-  vatAdminSvc,
+  vatAdminSvcP = undefined,
   shutdownZoeVat = () => {},
   feeIssuerConfig = defaultFeeIssuerConfig,
   zcfSpec = { name: 'zcf' },
   zoeBaggage = makeScalarBigMapStore('zoe baggage', { durable: true }),
 ) => {
-  let zoeService;
+  let zcfBundleCap;
+
+  function saveBundleCap() {
+    E.when(
+      Promise.all([vatAdminSvcP, getZcfBundleCap(zcfSpec, vatAdminSvcP)]),
+      ([vatAdminService, bundleCap]) => {
+        zcfBundleCap = bundleCap;
+
+        zoeBaggage.init('vatAdminSvc', vatAdminService);
+        zoeBaggage.init('zcfBundleCap', zcfBundleCap);
+      },
+    );
+  }
+
+  const setVatAdminService = lateVatAdminSvc => {
+    vatAdminSvcP = lateVatAdminSvc;
+    saveBundleCap();
```

Comment:
Should we be concerned about a potential race here in the event thta `setVatAdminService` is called again before `lateVatAdminSvc` is consumed?